### PR TITLE
[instrument] Support custom build tags in reporter

### PR DESCRIPTION
**Pull request by schallert**
**What this PR does / why we need it**:

The build reporter exposes helpful build metadata in the form of a
constantly updated gauge. However, the set of data it can expose is
fixed. This PR allows adding custom tags.

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

**Review by ryanhall07 — APPROVED** [review]

File: src/x/instrument/build.go [hunk 1]
```diff
@@ -131,13 +131,21 @@ func (b *buildReporter) Start() error {
...
+	tags := map[string]string{
 		"revision":      Revision,
 		"branch":        Branch,
 		"build-date":    BuildDate,
 		"build-version": Version,
 		"go-version":    goVersion,
-	})
+	}
+
+	if bt := b.opts.CustomBuildTags(); bt != nil {
```
Comment (ryanhall07) [inline src/x/instrument/build.go:142]: nit: don't need the if guard. you can range over a nil map

File: src/x/instrument/build.go [hunk 2]
```diff
@@ -131,13 +131,19 @@ func (b *buildReporter) Start() error {
...
 		"revision":      Revision,
 		"branch":        Branch,
 		"build-date":    BuildDate,
 		"build-version": Version,
 		"go-version":    goVersion,
-	})
+	}
+
+	for k, v := range b.opts.CustomBuildTags() {
+		tags[k] = v
```
Comment (arnikola) [inline src/x/instrument/build.go:143]: any worry of overriding existing tags?

**Review by arnikola — APPROVED** [review]
LGTM

File: src/x/instrument/build.go (hunk 2 above)
Comment (schallert) [reply src/x/instrument/build.go:143]: I can actually see cases in which we'd want to override. For example, the only way to really set `Version` etc. right now is by passing `ldflags`, and some users may have other ways they want to inject such metadata.

**Review by robskillington — APPROVED** [review]
LGTM

**Pull request merged**